### ISSUE TITLE
Disabling isolation has no effect on automatic mode

### ISSUE THREAD

**Issue by tokariu** [stoically/firefox-add-on-temporary-containers#581]
I noticed this behaviour lately on Firefox v100.

The temporarily "disable isolation" button in the adressbar dropdown does not have an effect. tabs are still opened in new random tabs (in automatic mode) despite having disabled isolation. There is no way to temporarily disable isolation now, the only way is to disable the addon completely in the firefox addon settings.

Previously the "disable isolation" setting (sometime before Firefox v100) from the adress bar dropdown menu worked perfectly fine and after applying it new tabs were all opened in the same (global) container.

It should be like that again.

**Comment by Gitoffthelawn** [comment]
My experience is different.  In Firefox 100, the "disable isolation" button still works for me.  Note that I do keep the Temporary Containers button in the address bar's overflow menu, and not in the address bar itself.  Maybe you could try that as a first step to see if it makes any difference?  It shouldn't make a difference, but then we will be comparing apples to apples.  (I would change my setup to test it, but I'm not on the correct system at this time.)

**Comment by tokariu** [comment]
@Gitoffthelawn I did some further tests and I think I can add some details to the problem:

it doesn't matter where the symbol resides (adress bar or overflow menu). I tried it both with a new firefox profile.

for some further testing:
if you have a bookmarks toolbar (below the adress bar) with some bookmarks on it and you click a bookmark with the middle-mouse-button to open it in a new tab, then this addon will now open each link (in a new tab) in a unique container tab DESPITE having the Isolation temporary disabled in the addon dropdown menu.

In previous versions the behaviour was as intended: middle-click opened bookmars were all using the same global container tabs until I re-enable Isolation in the dropdown menu - then each link opened that way will get their own unique container again (all this with automatic mode + domain+subdomain rule enabled of course).

the "disable isolation" mode seems to ignore "open link in new tab" and still isolates them. but it actually should disable isolation also in these cases.

**Comment by stoically** [comment]
Unfortunately can't reproduce. Can you try with a fresh profile / firefox installation?

**Comment by tokariu** [comment]
> Unfortunately can't reproduce. Can you try with a fresh profile / firefox installation?

I already did. And I just reproduced it again with a fresh firefox (v100) profile and a fresh "temporary containers" addon intallation.

Temporary Container Settings I've used:
- General / Automatic Mode
- General / Show popup when clicking the toolbar icon
- General / Random Container Color
- Isolation / Global / Navigation / Target Domain / Different from Tab Domain & Subdomains
- Advanced / General / Automatic Mode / "Instantly reopen new tabs in Temporary Containers (...)" (which is default setting)

everything else was left with default settings without touching.

**NOTICE:**
Most important is you have to distinguish between
a) clicking links on _webpages_ with the middle-mouse-button to open them in new tabs, and
b) middle-mouse-button-click (to open in new tab) on bookmarks at the _bookmarks-toolbar_ or any other _bookmarks-menu_ firefox provides.

because a) is working as intended, but b) is NOT.

in case of b) the addon seems to ignore the "disable isolation" setting.

**Comment by Gitoffthelawn** [comment]
@tokariu How are you able to click on *webpages* with the middle-mouse-button to open them in new tabs?  I tried that in a new profile and it was set by default to display a cursor to scroll the page with a mouse.

**Comment by tokariu** [comment]
> @tokariu How are you able to click on _webpages_ with the middle-mouse-button to open them in new tabs? I tried that in a new profile and it was set by default to display a cursor to scroll the page with a mouse.

Ah no, read carefully, I meant a) is to click _hyperLINKs on webpages_ with middle mouse button, not randomly somewhere on empty places of webpages.
also the middle-mouse-button is only synonym for "open in new tab", because that is exactly what firefox has assigned the middle mouse button to.

but don't focus solely on that, because opening links from webpages in new tabs is working as intended if isolation is disabled. the problem was that it is not working as intended when you open bookmarks in new tabs (as in: clicking bookmarks on the bookmarks-toolbar with the middle mouse button)

**Comment by stoically** [comment]
@tokariu 

> in case of b) the addon seems to ignore the "disable isolation" setting.

In this case TC behaves as expected, since clicking on bookmarks is handled by the automatic mode - not isolation. That's because middle-clicking on a bookmark loads a new tab in no container, which the automatic mode reloads into a temporary container.

Generally I'd agree that automatic mode should be part of isolation, but that would be a bigger refactoring. See also #397.

**Comment by tokariu** [comment]
> Generally I'd agree that automatic mode should be part of isolation, but that would be a bigger refactoring. See also #397.

indeed! I eagerly hope it'll be possible to implement once refactoring is going to happen.
